### PR TITLE
Fixed composter interactions on Forge

### LINKED ISSUE #4305: Hopperhock can't place compostable items inside composters
### Mod Loader

Forge

### Minecraft Version

1.19.2

### Botania version

Botania-1.19.2-437-FORGE

### Modloader version

Forge 1.19.2 - 43.1.65

### Modpack info

_No response_

### The latest.log file

-

### Issue description

imagen

### Steps to reproduce

place composter
put hopperhock on top / next to it
throw compostable item nearby 

### Other information

_No response_

### PR DESCRIPTION
**Pull request by TheRealWormbo**
Fixes #4305 and also allows Corporea Funnel to insert compostable items into composters. Hopperhocks can only insert when placed above the composter, similar to their slot interaction restrictions with furnaces or brewing stands.

### CODE REVIEW

File: Forge/src/main/java/vazkii/botania/forge/xplat/ForgeXplatImpl.java [hunk 1]
```diff
@@ -300,19 +303,27 @@ public boolean insertFluidIntoPlayerItem(Player player, InteractionHand hand, Fl
...
-		if (be == null) {
-			return toInsert;
+		LazyOptional<IItemHandler> cap = LazyOptional.empty();
+		if (be != null) {
+			cap = be.getCapability(ForgeCapabilities.ITEM_HANDLER, sideOfPos);
+		} else {
+			// check for vanilla interface, e.g. composter
+			var state = level.getBlockState(pos);
+			if (state.getBlock() instanceof WorldlyContainerHolder wch) {
+				cap = SidedInvWrapper.create(wch.getContainer(state, level, pos), sideOfPos)[0].cast();
```
Comment (williewillus) [inline Forge/src/main/java/vazkii/botania/forge/xplat/ForgeXplatImpl.java]: just call the constructor directly? `create` with only one side just constructs an array needlessly

File: Forge/src/main/java/vazkii/botania/forge/xplat/ForgeXplatImpl.java [hunk 2]
```diff
@@ -300,19 +303,27 @@ public boolean insertFluidIntoPlayerItem(Player player, InteractionHand hand, Fl
...
 	@Override
 	public ItemStack insertToInventory(Level level, BlockPos pos, Direction sideOfPos, ItemStack toInsert, boolean simulate) {
 		var be = level.getBlockEntity(pos);
-		if (be == null) {
-			return toInsert;
+		LazyOptional<IItemHandler> cap = LazyOptional.empty();
+		if (be != null) {
+			cap = be.getCapability(ForgeCapabilities.ITEM_HANDLER, sideOfPos);
+		} else {
+			// check for vanilla interface, e.g. composter
```
Comment (williewillus) [inline Forge/src/main/java/vazkii/botania/forge/xplat/ForgeXplatImpl.java]:
```suggestion
			// check vanilla interface for blocks not covered by forge capabilities, e.g. composter
```

**Review by williewillus — CHANGES_REQUESTED** [review]
nit

File: Forge/src/main/java/vazkii/botania/forge/xplat/ForgeXplatImpl.java (hunk 1 above)
Comment (TheRealWormbo) [reply Forge/src/main/java/vazkii/botania/forge/xplat/ForgeXplatImpl.java]: Good point. And technically it's even two arrays, one for the second parameter of the `create` call and another internally for the return value. Fixing with:
```suggestion
				cap = LazyOptional.of(() -> new SidedInvWrapper(wch.getContainer(state, level, pos), sideOfPos));
```

**Pull request merged**